arbstate/das_reader.go

```diff
@@ -41,30 +43,50 @@ type DataAvailabilityCertificate struct {
 	Sig         blsSignatures.Signature
 }
 
-func DeserializeDASCertFrom(buf []byte) (c *DataAvailabilityCertificate, bytesRead int, err error) {
+func DeserializeDASCertFrom(rd io.Reader) (c *DataAvailabilityCertificate, err error) {
+	r := bufio.NewReader(rd)
 	c = &DataAvailabilityCertificate{}
-	expectedCertSize := uintptr(1 + 32 + 8 + 8 + 96)
-	if uintptr(len(buf)) < expectedCertSize {
-		return nil, 0, fmt.Errorf("Can't deserialize DAS cert from smaller buffer (was %dB but should be %d)", uintptr(len(buf)), expectedCertSize)
+	expectedCertSize := 1 + 32 + 8 + 8 + 96
```

Comment:
It seems safer to not check the size here, but instead use `io.ReadFull` instead of `r.Read` below.  That way we'll still return an error on premature end-of-data, but we won't have to maintain the invariant that `expectedCertSize` is equal to the amount of data that we will need to read below.

Reply:
Thanks for showing me this! I made the change you suggested.